signalclient.go

```diff
@@ -348,18 +350,17 @@ func (c *SignalClient) handleResponse(res *livekit.SignalResponse) {
 func (c *SignalClient) readWorker() {
 	defer func() {
 		c.isStarted.Store(false)
-		if c.OnClose != nil {
-			c.OnClose()
-		}
+		c.conn.Store((*websocket.Conn)(nil))
```

Comment:
we cannot store nil into an atomic.Value; it'll panic: https://pkg.go.dev/sync/atomic#Value.Store

would be good to use typed atomic.Pointer instead.

Reply:
Casting `nil` to the concrete type does not cause a panic. Learnt that today. I can't find that post now, but here is a Go Playground example - https://go.dev/play/p/-O0qiTGJxJs>

But, I will change to look at `atomic.Pointer`

Reply:
Right, because a reference to an interface with a type but a nil object is not nil, which makes nil checks really confusing at time. Ask me how I know...

Reply:
How do you know @biglittlebigben 😊?

Reply:
Panic in prod at 3am. It's always a panic in prod at 3am...

Reply:
Using `atomic.Pointer` in this [commit](https://github.com/livekit/server-sdk-go/pull/351/commits/0e1c8fe0ebcd88598b16d75e0cde2ee29513ddc1).